File: app/src/main/java/com/lostincontext/ruledetails/items/PlaylistInEditScreenViewHolder.java
package com.lostincontext.ruledetails.items;


import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.lostincontext.commons.list.ViewHolder;
import com.lostincontext.data.playlist.Playlist;
import com.lostincontext.databinding.ItemPlaylistEditScreenBinding;

public class PlaylistInEditScreenViewHolder extends ViewHolder {


    private ItemPlaylistEditScreenBinding binding;

    public static PlaylistInEditScreenViewHolder create(LayoutInflater inflater,
                                                        ViewGroup parent) {
        ItemPlaylistEditScreenBinding itemBinding = ItemPlaylistEditScreenBinding.inflate(inflater,
                                                                                          parent,
                                                                                          false);
        return new PlaylistInEditScreenViewHolder(itemBinding);
    }


    public PlaylistInEditScreenViewHolder(ItemPlaylistEditScreenBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    public void bindTo(Playlist playlist) {
        binding.setPlaylist(playlist);
    }


}

File: app/src/main/java/com/lostincontext/ruledetails/items/PlaylistInEditScreenViewHolder.kt
package com.lostincontext.ruledetails.items


import android.view.LayoutInflater
import android.view.ViewGroup

import com.lostincontext.commons.list.ViewHolder
import com.lostincontext.data.playlist.Playlist
import com.lostincontext.databinding.ItemPlaylistEditScreenBinding

class PlaylistInEditScreenViewHolder(private val binding: ItemPlaylistEditScreenBinding) : ViewHolder(
        binding.root) {

    fun bindTo(playlist: Playlist) {
        binding.playlist = playlist
    }

    companion object {

        fun create(inflater: LayoutInflater,
                   parent: ViewGroup): PlaylistInEditScreenViewHolder {
            val itemBinding = ItemPlaylistEditScreenBinding.inflate(inflater,
                                                                    parent,
                                                                    false)
            return PlaylistInEditScreenViewHolder(itemBinding)
        }
    }


}
